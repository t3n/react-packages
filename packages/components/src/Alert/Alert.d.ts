import React from 'react';
import { SpaceProps } from 'styled-system';
export declare type AlertStatus = 'success' | 'notice' | 'warning' | 'error';
interface AlertProps extends SpaceProps {
    status: AlertStatus;
}
declare const AlertText: React.FC;
declare const Alert: React.FC<AlertProps>;
export { Alert, AlertText };
