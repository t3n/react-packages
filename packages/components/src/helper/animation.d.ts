import { ThemeProps } from '@t3n/theme';
export declare const backgroundAnimation: import("styled-components").FlattenInterpolation<import("styled-components").ThemedStyledProps<ThemeProps, any>>;
