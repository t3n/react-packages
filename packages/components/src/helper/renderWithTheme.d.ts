import 'jest-styled-components';
import React from 'react';
import { RenderOptions } from '@testing-library/react';
import { Theme } from '@t3n/theme';
interface OptionalTheme {
    theme?: Theme;
}
export declare const renderWithTheme: (ui: React.ReactElement, { theme, ...options }: RenderOptions & OptionalTheme) => import("@testing-library/react").RenderResult<typeof import("@testing-library/dom/types/queries"), HTMLElement>;
export {};
