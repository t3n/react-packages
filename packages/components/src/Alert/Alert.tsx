import React from 'react';
import styled from 'styled-components';
import { border, space, SpaceProps, variant } from 'styled-system';

import { ThemeProps } from '@t3n/theme';

import Box from '../Box';
import Text from '../Text';

export type AlertStatus = 'success' | 'notice' | 'warning' | 'error';

export interface AlertProps extends SpaceProps {
  status: AlertStatus;
}

const AlertBox = styled(Box)<{ status: AlertStatus }>`
  ${({ theme }: ThemeProps) =>
    border({
      theme,
      borderColor: 'shades.grey42',
      borderWidth: '1px 2px 4px 1px',
      borderStyle: 'solid',
      borderRadius: theme.border.radii[1],
    })}

  ${variant({
    prop: 'status',
    variants: {
      success: {
        bg: 'feedback.success',
      },
      notice: {
        bg: 'feedback.notice',
      },
      warning: {
        bg: 'feedback.warn',
      },
      error: {
        bg: 'feedback.error',
        color: 'shades.white',
      },
    },
  })}
  ${space};
`;

// todo export AlertIcon as soon as we do have an IconComponent

export const AlertText: React.FC = ({ children }) => {
  return (
    <Text m={0} width="100%">
      {children}
    </Text>
  );
};

const Alert: React.FC<AlertProps> = ({ status, children, ...rest }) => {
  return (
    <AlertBox
      display="flex"
      alignItems="center"
      status={status}
      p={3}
      {...rest}
    >
      {children}
    </AlertBox>
  );
};

export default Alert;
