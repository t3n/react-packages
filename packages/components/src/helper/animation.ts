import { css, keyframes } from 'styled-components';

import { ThemeProps } from '@t3n/theme';

const backgroundAnimationKeyframes = ({ theme }: ThemeProps) => keyframes`
  0% {
    background-color: ${theme.colors.shades.grey204}
  }

  50% {
    background-color: ${theme.colors.shades.grey244}
  }

  100% {
    background-color: ${theme.colors.shades.grey204}
  }
`;

// eslint-disable-next-line import/prefer-default-export
export const backgroundAnimation = css`
  animation: ${backgroundAnimationKeyframes} 2s linear infinite;
`;
