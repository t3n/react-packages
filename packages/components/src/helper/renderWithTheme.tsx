/* eslint-disable import/no-extraneous-dependencies */
import 'jest-styled-components';

import React from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';

import { Theme, theme as originalTheme } from '@t3n/theme';

interface OptionalTheme {
  theme?: Theme;
}

// eslint-disable-next-line import/prefer-default-export
export const renderWithTheme = (
  ui: React.ReactElement,
  { theme, ...options }: RenderOptions & OptionalTheme
) => {
  // merge over default theme

  const mergedTheme = { ...originalTheme, ...theme };

  const Wrapper: React.FC = ({ children }) => (
    <ThemeProvider theme={mergedTheme}>{children}</ThemeProvider>
  );

  return render(ui, { wrapper: Wrapper, ...options });
};
