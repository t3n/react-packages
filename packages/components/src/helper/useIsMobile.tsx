import { Theme } from '@t3n/theme';
import { useCallback, useContext, useEffect, useState } from 'react';
import { ThemeContext } from 'styled-components';

const useIsMobile = (): boolean => {
  const [isMobile, setIsMobile] = useState(false);

  const theme: Theme = useContext(ThemeContext);
  const fontSize = window
    .getComputedStyle(window.document.body)
    .getPropertyValue('font-size');
  const mobileBreakpoint =
    parseInt(theme.breakpoints[1], 10) * parseInt(fontSize, 10);

  const testIsMobile = useCallback(() => {
    if (window.matchMedia(`(max-width: ${mobileBreakpoint}px)`).matches) {
      setIsMobile(true);
    } else {
      setIsMobile(false);
    }
  }, [mobileBreakpoint]);

  useEffect(() => {
    testIsMobile();
    window.addEventListener('resize', testIsMobile);

    return () => {
      window.removeEventListener('resize', testIsMobile);
    };
  }, [testIsMobile]);

  return isMobile;
};

export default useIsMobile;
