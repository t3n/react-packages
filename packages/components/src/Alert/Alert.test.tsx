import React from 'react';
import { renderWithTheme } from '../helper/renderWithTheme';
import { Alert, AlertText } from './Alert';

test('Alert Status matches Snapshot', () => {
  const { container } = renderWithTheme(
    <>
      <Alert status="success" m={2}>
        <AlertText>Dies ist eine Erolgsmeldung</AlertText>
      </Alert>
      <Alert status="notice" m={2}>
        <AlertText>Merke dir dies und das!</AlertText>
      </Alert>
      <Alert status="warning" m={2}>
        <AlertText>Dies ist eine Warnung!</AlertText>
      </Alert>
      <Alert status="error" m={2}>
        <AlertText>So geht das nicht. Error!</AlertText>
      </Alert>
    </>,
    {}
  );

  expect(container).toMatchSnapshot();
});
