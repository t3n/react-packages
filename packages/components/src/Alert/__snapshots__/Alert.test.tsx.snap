// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert Status matches Snapshot 1`] = `
.c0 {
  padding: 16px;
  margin: 8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  font-family: -apple-system,system-UI,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica Neue,Arial,sans-serif;
  font-size: 1rem;
  line-height: 1.5em;
  margin: 0;
  width: 100%;
}

.c1 {
  border-color: #2a2a2a;
  border-width: 1px 2px 4px 1px;
  border-style: solid;
  border-radius: 4px;
  background-color: #3DFF9F;
  padding: 16px;
  margin: 8px;
}

.c3 {
  border-color: #2a2a2a;
  border-width: 1px 2px 4px 1px;
  border-style: solid;
  border-radius: 4px;
  background-color: #56FEFF;
  padding: 16px;
  margin: 8px;
}

.c4 {
  border-color: #2a2a2a;
  border-width: 1px 2px 4px 1px;
  border-style: solid;
  border-radius: 4px;
  background-color: #ffef4f;
  padding: 16px;
  margin: 8px;
}

.c5 {
  border-color: #2a2a2a;
  border-width: 1px 2px 4px 1px;
  border-style: solid;
  border-radius: 4px;
  background-color: #f9423a;
  color: #fff;
  padding: 16px;
  margin: 8px;
}

@media screen and (min-width:30rem) {
  .c2 {
    line-height: 1.5em;
  }
}

@media screen and (min-width:48rem) {
  .c2 {
    line-height: 1.666em;
  }
}

<div>
  <div
    class="c0 c1"
    display="flex"
  >
    <p
      class="c2"
      width="100%"
    >
      Dies ist eine Erolgsmeldung
    </p>
  </div>
  <div
    class="c0 c3"
    display="flex"
  >
    <p
      class="c2"
      width="100%"
    >
      Merke dir dies und das!
    </p>
  </div>
  <div
    class="c0 c4"
    display="flex"
  >
    <p
      class="c2"
      width="100%"
    >
      Dies ist eine Warnung!
    </p>
  </div>
  <div
    class="c0 c5"
    display="flex"
  >
    <p
      class="c2"
      width="100%"
    >
      So geht das nicht. Error!
    </p>
  </div>
</div>
`;
